contracts/contracts/utils/Initializable.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity ^0.8.0;
+
+abstract contract Initializable {
+    /**
+     * @dev Indicates that the contract has been initialized.
+     */
+    bool private initialized;
+
+    /**
+     * @dev Indicates that the contract is in the process of being initialized.
+     */
+    bool private initializing;
+
+    /**
+     * @dev Modifier to protect an initializer function from being invoked twice.
+     */
+    modifier initializer() {
+        require(
+            initializing || !initialized,
+            "Initializable: contract is already initialized"
+        );
+
+        bool isTopLevelCall = !initializing;
+        if (isTopLevelCall) {
+            initializing = true;
+            initialized = true;
+        }
+
+        _;
+
+        if (isTopLevelCall) {
+            initializing = false;
+        }
+    }
+
+    uint256[50] private ______gap;
```

Comment:
Is this gap new? Is this going to throw off every storage slot after in the vault and in OUSD?

Reply:
See the PR comment for an explanation

Reply:
Great.